io/local.go

```diff
@@ -30,6 +31,18 @@ func (LocalFS) Open(name string) (File, error) {
 	return os.Open(strings.TrimPrefix(name, "file://"))
 }
 
+func (LocalFS) Create(name string) (FileWriter, error) {
+	filename := strings.TrimPrefix(name, "file://")
+	if err := os.MkdirAll(filepath.Dir(filename), 0755); err != nil {
```

Comment:
are these modes correct? `0755` for Mkdir and `0644` for WriteFile

Reply:
I kinda just picked the modes out of a hat for defaults as they made sense, perhaps they should both be 0644?

Reply:
i just saw magic numbers so i wanted to double check, not sure what the default is either

https://grep.app/search?q=os.MkdirAll%28&filter[lang][0]=Go

there are some `0777` and `0755`s

Reply:
Is there a common default used by pyiceberg or java-iceberg when creating the metadata directories or other file directory creation?

Reply:
ok so for pyiceberg [fsspec's LocalFileSystem](https://github.com/apache/iceberg-python/blob/3981e620f4560e3d13bff3d062bd3a89e39971d0/pyiceberg/io/fsspec.py#L120)

the underlying implementation calls [`os.makedirs`](https://github.com/fsspec/filesystem_spec/blob/216885a806844e6c69854b226210055c46850c0e/fsspec/implementations/local.py#L52) which [defaults to `0777`](https://docs.python.org/3/library/os.html#os.makedirs)

Reply:
Ok, I'll update both of these to 0777 then